solr/core/src/java/org/apache/solr/request/SolrRequestInfo.java

```diff
@@ -210,6 +216,32 @@ public void addCloseHook(Closeable hook) {
     }
   }
 
+  /**
+   * This call creates the QueryLimits object and any required implementations of {@link
+   * org.apache.lucene.index.QueryTimeout}. Any code before this call will not be subject to the
+   * limitations set on the request. Note that calling {@link #getLimits()} has the same effect as
+   * this method.
+   *
+   * @see #getLimits()
+   */
+  public void initQueryLimits() {
```

Comment:
why public; seems internal

Reply:
had been called elsewhere as getLimits(), agree can be private now.